test/core.jl

```diff
@@ -4822,6 +4822,17 @@ end
 @test f14893() == 14893
 @test M14893.f14893() == 14893
 
+# issue #18725
+@test_nowarn begin
```

Comment:
I've pushed a workaround, and verified that it is not printing any warning that fails to get captured.  (The issue of when the method-override warning gets printed seems unrelated to this PR.)